### PR TITLE
[Infinity-1311] test output stderr

### PR DESCRIPTION
**Pull request by jsrodman**
After a few failed experiments, the working pattern is to just ask python not to buffer output.  Since test.py is already running run_tests.py with the same file as fd 1 and 2, this should provide a reliable linear stream of events.   That's a very small change.

The much larger linecount change is the move from print to a dedicated "this is test output" function.  I'm not married this.  But I wanted to move away from print inline everywhere since that's a growing problem.  

I can also decouple the test.py fix, though given the pace of my reviews and merges I'm not excited about it.

### CODE REVIEW

**Comment by nickbp** [comment]
Maybe shorten usage via either or both:
- name the function `p`?
- import in tests against an alias like `p`?

**Comment by jsrodman** [comment]
@nickbp I don't like copying references out of module namespaces; I feel it leads to less clarity as to what code is being called and I've experience the pain of multiple copies of module implementations in memory at times.

The name certainly could be shorter.  out() ?

File: testing/sdk_utils.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+    #    del parent
+    #name = inspect.getmodulename(parent_filename)
+    #out = "{current_time} {name} {msg}\n".format(current_time=current_time,
+    #                                             name=name,
+    #                                             msg=msg)
+    #sys.stderr.write(out)
+
+
 def gc_frameworks():
+    '''Reeclaims private agent disk space consumed by Mesos but not yet garbage collected'''
```
Comment (loren) [inline testing/sdk_utils.py]: reclaims

File: test.py [hunk 2]
```diff
@@ -505,7 +510,8 @@ def start_test_background(framework, cluster, repo_root):
                 framework.name)
     _setup_strict(framework, cluster, repo_root)
 
-    logger.info("Launching shakedown for %s", framework.name)
+    logger.info("Launching shakedown in background for %s", framework.name)
+    logger.debug("stub_universe:{framework.stub_universe_url} cluster_url:{cluster.url} authtoken:{cluster.auth_token}".format(locals()))
```
Comment (loren) [inline test.py]: If you are keeping this debug and the one on line 546, please pick one style of formatting that string

Comment (jsrodman) [inline test.py]: Sure, i'll keep the one that actually works.

**Comment by jsrodman** [comment]
btw, the failures are of course me cancelling redundant jobs.

**Comment by benclarkwood** [comment]
I like `out()`

**Comment by benclarkwood** [comment]
And also don't like removing module namespacing. Namespacing is one of my favorite parts of Go

**Comment by jsrodman** [comment]
retest this please

**Comment by benclarkwood** [comment]
@jsrodman has this fixed what you wanted it to fix?

**Comment by jsrodman** [comment]
@benclarkwood it's "fixed enough" when being run by test.py with output redirection.  

Without redirection, there are still open questions about whatever receiver system is doing with the streams.  How does jenkins pretend to be a terminal? I couldn't say.

Initially i'd wanted everything going to stderr with timestamps, but experimentation showed that is not achievable yet; shakedown would need significant change.

**Comment by jsrodman** [comment]
Unrelatedly,  frameworks.cassandra.tests.test_recovery_shutdown.test_shutdown_host_test failed.

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
well, kafka hit an SSL error.  Okay whatever.

**Comment by jsrodman** [comment]
retest this please

**Pull request merged**